src/tinyformat.h

```diff
@@ -1061,6 +1061,23 @@ std::string format(const std::string &fmt, const Args&... args)
     return oss.str();
 }
 
+/** Variant of tfm::format that aborts when an formatting exception happens instead
+ * of raising an exception.
+ * Use this function in no-except context, but only when you are 100% sure that the
+ * format string contains no mistakes.
+ */
+template<typename... Args>
+std::string format_noexcept(const std::string &fmt, const Args&... args) noexcept
+{
+    std::ostringstream oss;
+    try {
+        format(oss, fmt.c_str(), args...);
+    } catch(...) {
+        abort();
```

Comment:
Instead of aborting what about returning `Unable to format: "…"` in case of a format string mistake?

Perhaps with an `ASSUME(false);` to make it abort if `--enable-debug` (674f9d59ba4c1a1d4662e79467cc417a2988cf15 in #16136 gives us `ASSUME(…)`).

Reply:
This is intended for use in utility functions, in neither of these cases it's acceptable to end up with "Unable to format: " in the string! That's much worst than to just terminate the process.

No opinion on how to terminate the process, though. I did not use `assert(0)` because I don't want to rely on assert() not being optimized out, if in the future `NDEBUG` will be allowed.

Reply:
OK, I see your point :)

Reply:
OK, I see your point :)

Reply:
Oh, sorry: I now re-read the diff and understand that the use of `format_noexcept` is extremely limited. Makes perfect sense. Sorry for the misunderstanding.

Reply:
I think you don't have to call `std::abort` in a throwing function that is marked `noexcept(true)` will `std::terminate` the program anyway, no?

Reply:
@MarcoFalke correct. I was not sure to make this explicit or not.

Reply:
Yes, I believe that the program will always abort, even if someone sets a terminate handler that does not abort: https://en.cppreference.com/w/cpp/error/terminate_handler

So it should be fine to remove this and add a comment instead.

Reply:
Done